### PR TITLE
Implement CodexMessageFormatter for Linear comment formatting

### PR DESCRIPTION
**Pull request by cyrusagent**
## Summary

Implements the `CodexMessageFormatter` class that transforms Codex CLI tool outputs into human-readable markdown for Linear comments. This is part 4 of 7 in the Codex runner implementation stack (CYPACK-525).

## Implementation

**CodexMessageFormatter class** (`packages/codex-runner/src/formatter.ts`):
- Implements all 4 methods of the `IMessageFormatter` interface
- Handles Codex-specific tools: `command_execution`, `file_change`, `reasoning`
- Formats MCP tools with `server:tool` pattern
- Language detection for syntax highlighting (25+ languages)
- Smart truncation for long outputs
- Todo formatting with status emojis (✅ 🔄 ⏳)
- Error formatting with appropriate styling
- Circular reference handling

**Key Features:**
- **Command execution**: Displays shell commands with bash syntax highlighting
- **File changes**: Shows file paths, change types, and content with language detection
- **Reasoning**: Formats as blockquotes for visual distinction
- **MCP tools**: Formats as `server:tool_name` pattern
- **Error handling**: Code blocks with error styling and console logging
- **Truncation**: Intelligently truncates long content with ellipsis

## Testing

Comprehensive test suite with 35 test cases covering:
- All formatter methods (formatTodoWriteParameter, formatToolParameter, formatToolActionName, formatToolResult)
- Tool-specific formatting verification
- Edge cases (empty data, errors, circular references)
- Language detection across 25+ file types
- Output truncation verification

**Test Results:**
- ✅ 135 tests passing (35 new formatter tests + existing tests)
- ✅ Linting clean
- ✅ TypeScript types valid
- ✅ Build successful

## Stack Position

**Position 4 of 7** in Graphite stack
- **Previous**: CYPACK-524 (Codex adapters)
- **Dependencies**: Tool name conventions from adapters

## Related

- Issue: CYPACK-525
- Stack: Codex runner implementation

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

**Comment by cyrusagent** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/ceedaragents/cyrus/606?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#612** <a href="https://app.graphite.com/github/pr/ceedaragents/cyrus/612?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#611** <a href="https://app.graphite.com/github/pr/ceedaragents/cyrus/611?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#610** <a href="https://app.graphite.com/github/pr/ceedaragents/cyrus/610?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#609** <a href="https://app.graphite.com/github/pr/ceedaragents/cyrus/609?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#608** <a href="https://app.graphite.com/github/pr/ceedaragents/cyrus/608?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#607** <a href="https://app.graphite.com/github/pr/ceedaragents/cyrus/607?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#606** <a href="https://app.graphite.com/github/pr/ceedaragents/cyrus/606?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/ceedaragents/cyrus/606?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#605** <a href="https://app.graphite.com/github/pr/ceedaragents/cyrus/605?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#604** <a href="https://app.graphite.com/github/pr/ceedaragents/cyrus/604?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#603** <a href="https://app.graphite.com/github/pr/ceedaragents/cyrus/603?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>cyrus</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/ceedaragents?ref=merge-queue-instructions-in-stack-comment&prId=606).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/codex-runner/src/formatter.ts [hunk 1]
```diff
@@ -1 +1,409 @@
...
+					}
+					break;
+				}
+
+				case "reasoning": {
+					// Truncate reasoning text for parameter display
+					const text = getString(toolInput, "text");
+					if (text) {
+						return truncate(text, 100);
+					}
```
Comment (PaytonWebber) [inline packages/codex-runner/src/formatter.ts:192]: Why truncate here?
